docs/design/coreclr/botr/data_contracts_for_coreclr.md

```diff
@@ -0,0 +1,430 @@
+# Data Contracts for CoreCLR, and the data contract stream
+
+Data contracts represent the manner in which a tool which is not the runtime can reliably understand and observe the behavior of the runtime. Contracts are defined by their documentation, and the runtime describes what contracts are applicable to understanding that runtime.
+
+## Data Contract Stream
+The physical layout of this stream is not defined in this document, but its practical effects are.
+
+The data contract stream has a set of records of the following forms.
+
+### Global Values
+Global values which can be of types (int8, uint8, int16, uint16, int32, uint32, int64, uint64, pointer, unicode string)
```

Comment:
I think we also need `size_t`+`ptrdiff_t`/`nint`+`nuint` types - there are places in the runtime where we have non-pointer pointer-sized data.

Reply:
Sure. I think I would do nint/nuint instead of size_t/ptrdiff_t.